docs/migration.rst

```diff
@@ -74,7 +74,17 @@ AttributeError in Shapely 2.0.
 **How do I update my code?** There is no direct alternative for adding custom
 attributes to geometry objects. You can use other Python data structures such as
 (GeoJSON-like) dictionaries or GeoPandas' GeoDataFrames to store attributes
-alongside geometry features. 
+alongside geometry features.  
+You can also create a costum object holding a reference to a shapely geometry object, in addition to the custom attribute(s)::
+
+    from shapely.geometry import LineString
+    
+    class CustomObject():
```

Comment:
An alternative could also be to use a dataclass like:

```
from shapely.geometry.base import BaseGeometry
from dataclasses import dataclass

@dataclass
class CustomFeature:
    attribute1: int
    geometry: BaseGeometry
```

(and because there are many ways to do this, I am not fully sure to what extent we should give a very explicit code example, versus explaining in words)

Reply:
Hi Joris, well the data-class is special case of the CustomObject I propose in the end, right ? it still illustrates the alternative of holding a reference to a geometry object.

I agree they are many ways to do it, but for having been confronted with the issue, a code example can save some time and is indeed more explicit if one is not familiar with the actual terms (like "using an attribute holding a reference to a geometry object" might not be obvious to everyone).

We typed the example code already actually (thanks for the tip about dataclass decorator btw, did not know about it) so why not put both and let the user decide what they want to use. This might not be the best solution but it's short, functional and does not rely on any other package.